cluster-agent/controllers/argoproj.io/application_controller_test.go

```diff
@@ -566,6 +575,229 @@ var _ = Describe("Application Controller", func() {
+				Spec_field:              "{}",
+				Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+			}
+
+			applicationStateDB := &db.ApplicationState{
+				Applicationstate_application_id: applicationDB.Application_id,
+			}
+
+			By("Create Application in Database")
+			err = reconciler.DB.CreateApplication(ctx, applicationDB)
+			Expect(err).To(BeNil())
+
+			By("Verify SyncError is not present in database")
+			err = reconciler.DB.GetApplicationStateById(ctx, applicationStateDB)
+			Expect(err).ToNot(BeNil())
+			Expect(db.IsResultNotFoundError(err)).To(BeTrue())
+
+			By("Create a new ArgoCD Application")
+			err = reconciler.Create(ctx, guestbookApp)
+			Expect(err).To(BeNil())
+
+			By("Call reconcile function")
+			result, err := reconciler.Reconcile(ctx, newRequest(namespace, name))
+			Expect(err).To(BeNil())
+			Expect(result).ToNot(BeNil())
+
+			err = reconciler.DB.GetApplicationStateById(ctx, applicationStateDB)
+			Expect(applicationStateDB.ReconciledState).ToNot(BeNil())
+			Expect(err).To(BeNil())
+
+			// Checking for new values in database object
+			for _, syncError := range guestbookApp.Status.Conditions {
+				Expect(applicationStateDB.SyncError).To(Equal(syncError.Message))
+				Expect(syncError.Type).To(Equal("SyncError"))
+			}
+		})
+
+		It("Test to verify the conditions.Type is not equal to 'SyncError' then db field should be empty", func() {
+			By("Close database connection")
+			defer dbQueries.CloseDatabase()
+			defer testTeardown()
+			Expect(err).To(BeNil())
+
+			ctx = context.Background()
+
+			guestbookApp := &appv1.Application{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Application",
+					APIVersion: "argoproj.io/v1alpha1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+					Labels: map[string]string{
+						dbID: "test-my-application",
+					},
+				},
+				Spec: appv1.ApplicationSpec{
+					Source: appv1.ApplicationSource{
+						Path:           "guestbook",
+						TargetRevision: "HEAD",
+						RepoURL:        "https://github.com/argoproj/argocd-example-apps.git",
+					},
+					Destination: appv1.ApplicationDestination{
+						Namespace: "guestbook",
+						Server:    "https://kubernetes.default.svc",
+					},
+					Project: "default",
+				},
+				Status: appv1.ApplicationStatus{
+					Health: appv1.HealthStatus{
+						Status: "Healthy",
+					},
+					Sync: appv1.SyncStatus{
+						Status: "Synced",
+						ComparedTo: appv1.ComparedTo{
+							Source: appv1.ApplicationSource{
+								Path:           "test-path",
+								RepoURL:        "test-url",
+								TargetRevision: "test-branch",
+							},
+							Destination: appv1.ApplicationDestination{
+								Namespace: "test-namespace",
+								Name:      "managed-env-123",
+							},
+						},
+					},
+					Conditions: []appv1.ApplicationCondition{
+						{
+							Type:               "test-error",
```

Comment:
Rather than using `test-error`, you can use one of the other error types, like `ApplicationConditionUnknownError`